### ISSUE TITLE
Rule idea: String call instead of template literal

### ISSUE THREAD

**Issue by krutoo** [eslint/eslint#11998]
Hi. I have an idea for the rule that will make reading the code more convenient. Instead of template literal with one value and without regular string around, `String` call looks more clearly.

Example of bad code:
```javascript
// prefer-string-call: "error"
const stringCount = `${getCount()}`; // Expected call of String() instead of template literal
```

Example of good code:
```javascript
// prefer-string-call: "error"
const stringCount = String(getCount);
```

**Comment by platinumazure** [comment]
Hi @krutoo, thanks for the issue.

Personally, this feels like a subjective preference and I could see people going either way on this. Also, this covers a pretty small subset of strings. So I'm mildly opposed to adding this to core, but I will go along with the team consensus.

It sounds like this would be a great case for a custom rule for your codebase.

**Comment by kaicataldo** [comment]
Agreed with @platinumazure. ESLint core has a lot of rules and at this point we have a really high bar for adding new rules (since we have to maintain them!).

**Comment by krutoo** [comment]
I understand that this rule is not definitely suitable for being in the core of ESLint. But i do not know how to be in such a situation. Maybe I need to expand the idea and the community will help me.

**Comment by kaicataldo** [comment]
We have documentation on creating custom rules [here](https://eslint.org/docs/developer-guide/working-with-rules). Please also feel free to stop by our [chat](https://gitter.im/eslint/eslint) if you have any questions!

**Comment by mysticatea** [comment]
Actually, [no-implicit-coercion](https://eslint.org/docs/rules/no-implicit-coercion) rule is the best place to check this. That rule warns `"" + foo` to enforce to use `String(foo)`. This is very similar.

**Comment by platinumazure** [comment]
@mysticatea Do you want to champion this?

**Comment by mysticatea** [comment]
Yes if the issue updated -- I don't think I want to add a new rule.

**Comment by ljharb** [comment]
When template literals are available, nobody should be using `String()`, because `global.String = function () { return 42; }` will break that code.

The reason `'' + foo` is bad is because it calls `valueOf` first, not because it's implicit (it's not) - a template literal, like `String()`, calls `toString` first, which makes it the best approach hands down.

This rule should not exist.

**Comment by krutoo** [comment]
@ljharb obviously, this is not the case, at least because the style of the code may be preferable to security.

In addition, already now there is a rule "no-implicit-coercion" which also calls for using `String` (with some settings).

**Comment by krutoo** [comment]
Thank you all for reviews. I think this rule must be a custom, not in core of ESLint, at least in this form.

**Comment by ljharb** [comment]
Correctness and robustness should always win out over style; as I said, using `String` is more correct than concatenating to a string; and using a template literal is equally correct and also more robust.

**Comment by krutoo** [comment]
@ljharb Following this logic, you can make rule settings to choose between style and security.

Correctness must be justified. It makes no sense to try to make JS in the browser safe in such little things.

The issue of security is wider and most likely should be solved by introducing new JS modes like 'use strict' which would prevent the redefinition of global variables. 

Without global variables you simply cannot use JS.

**Comment by ljharb** [comment]
This is getting off topic for the issue, so this will be the last comment I make on the topic :-)

It's entirely possible to make JS code fully robust, as long as it has the chance to run first, before any untrusted code. You certainly can do something like `const $String = String;` at the top of your module, and then reference `$String(x)` in your code, and that will be robust - but at that point I suspect the style benefit drops quite a bit.

Separately, JS will never add a new mode ever again, and it's forever going to be up to your app code to protect itself against later-run code doing malicious things.

What makes no sense to me is *not* trying to make your website robust ¯\\\_(ツ)_/¯ but to each their own.

**Comment by krutoo** [comment]
@ljharb I fully agree: priorities depend on the project.

About the special variable like `$String`: perhaps this is capable of and should be done by the transpiler or other build tool. Then the code style will remain clean. 

Thank you for review.
